doc/source/custom_directives.py

```diff
@@ -50,11 +54,6 @@ def feedback_form_url(project, page):
     )
 
 
-def update_context(app, pagename, templatename, context, doctree):
```

Comment:
could you explain how is this change related? why the `update_context` function needs to be removed and inlined now?

Reply:
> could you explain how is this change related?

Strictly speaking this isn't a necessary change, but right now we have two functions which run on `html-page-context`, both of which add things to the context. This is confusing for people who want to understand the docs build system, and since I was touching the context anyway to add the pygments highlighter I thought it would be a good time to fix this antipattern.

Reply:
maybe separate this change into another PR?

Reply:
Sure, will do.